lesson15/serkan/script.js

```diff
@@ -0,0 +1,134 @@
+const anotherString = 'A B C D E';
+console.log(anotherString.slice(-5, -3));
+console.log(anotherString.substring(-5, -3)); // Does not give us the same result as Slice
+
+// 5. Trim Whitespace
+// Clean up user input by removing unnecessary spaces.
+const messyInput = '   messy input  ';
+const withoutSpaces = messyInput.trim();
+console.log(withoutSpaces);
+
+// 6. Check if String Contains Substring
+// Verify if a URL contains "https".
+const url = 'HTTPS://example.com';
+const protocol = 'https';
+console.log(url.toLowerCase().includes(protocol));
+
+// 7. Find Character at Index
+// Find the initials of the name.
+const firstName = 'D John Doe D ';
+console.log(firstName.indexOf('D', 12), firstName.indexOf('J'));
+
+if (firstName.indexOf('Jane') === -1) {
+	console.log("Name isn't Jane");
+}
+
+// 8. Find Index of Substring (H)
+// Find the position of a word 'JavaScript' in a sentence.
+const sentenceWithAWordToFind =
+	"The object-oriented nature of JavaScript is important to understand if you want to go further with your knowledge of the language and write more efficient code, therefore we've provided this module to help you.";
+
+const findTheJavascript=sentenceWithAWordToFind.indexOf("JavaScript");
+console.log(findTheJavascript);
+// 9. Extract a Substring
+// Extract the domain from an email address.
+const email = 'user-name@google.com';
+const startIndexOfDomain = email.indexOf('@') + 1;
+const domain = email.substring(startIndexOfDomain);
+
+console.log(domain);
+// 10. Replace Substring
+// Censor (replace 'bad' with 'good') a word in a sentence.
+const sentence = 'This is a bad example bad.';
+console.log(sentence.replace('bad', 'good'));
+
+// 11. Repeat a String (H)
+// Create a visual separator for console output. Expected result: "--------------------"
+const separator = "-".repeat(10);
+console.log(separator);
+
+// 12. Check if String Starts With (H)
+// Check if a file name starts with a specific prefix. Return true or false.
+const fileName = 'report_2024.pdf';
+console.log(fileName.startsWith('rep'));
+
+// 13. Check if String Ends With (H)
+// Check if a file name ends with a specific extension. Return true or false.
+console.log(fileName.endsWith('fd'));
+
+// 14. Pad Start of String (H)
+// Pad a number with zeros using padStart() function to ensure it has at least 5 digits.
+const number = '42';
+const startsWithZero=number.padStart(5,'0');
+console.log(startsWithZero);
+
+// 15. Pad End of String (H)
+// Pad a text with spaces to align it to the right. Expected result: "Align     "
+const text = 'Align';
+const addWithSpaces=text.endsWith(5,' ')
+console.log(addWithSpaces);
+
+// 16. Split a String (H)
+// Split a sentence into words.
+const sentence2 = 'Learning JavaScript is fun';
+const splitSentence=sentence2.split(' ');
+console.log(splitSentence);
+
+// 17. Convert String to Array of Characters (H)
+// Convert a string into an array of its characters. Use sentence2 variable from exercise above.
+
```

Comment:
To split a string into an array of characters, use split method with empty string as a parameter.
```suggestion
console.log(sentence2.split(''));
```